module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,58 @@
+#include "compression.hpp"
+#include <algorithm>
+#include <iostream>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& imgToCompress) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressedImg;
+ 
+    std::for_each(imgToCompress.begin(), imgToCompress.end(), [&compressedImg](const auto& row) {
+        uint8_t pixCount = 0;
+        for (auto it = row.begin(); it != row.end(); ++it) {
+            if (it == row.begin() || *it == *(it - 1)) {
+                ++pixCount;
+            } 
+            else {
+                compressedImg.push_back(std::make_pair(*(it - 1), pixCount));
+                pixCount = 1;
+            }
+
+            if (it == row.end() - 1) {
```

Comment:
prev should do the trick :)

Reply:
You mean `std::prev` ? Good point, thanks!

Reply:
+1